pkg/server/plugin/keymanager/awskms/fetcher.go

```diff
@@ -0,0 +1,137 @@
+package awskms
+
+import (
+	"context"
+	"path"
+	"strings"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/hashicorp/go-hclog"
+	keymanagerv0 "github.com/spiffe/spire/proto/spire/plugin/server/keymanager/v0"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type keyFetcher struct {
+	log         hclog.Logger
+	kmsClient   kmsClient
+	serverID    string
+	trustDomain string
+}
+
+func (kf *keyFetcher) fetchKeyEntries(ctx context.Context) ([]*keyEntry, error) {
+	var keyEntries []*keyEntry
+	var keyEntriesMutex sync.Mutex
+	paginator := kms.NewListAliasesPaginator(kf.kmsClient, &kms.ListAliasesInput{Limit: aws.Int32(100)})
+	g, ctx := errgroup.WithContext(ctx)
+
+	for {
+		aliasesResp, err := paginator.NextPage(ctx)
+		switch {
+		case err != nil:
+			return nil, status.Errorf(codes.Internal, "failed to fetch aliases: %v", err)
+		case aliasesResp == nil:
+			return nil, status.Errorf(codes.Internal, "failed to fetch aliases: nil response")
+		}
+
+		kf.log.Debug("Found aliases", "num_aliases", len(aliasesResp.Aliases))
+
+		for _, alias := range aliasesResp.Aliases {
+			switch {
+			case alias.TargetKeyId == nil:
+				// this means something external to the plugin created the alias, without associating it to a key.
+				// it should never happen with CMKs.
+				return nil, status.Errorf(codes.FailedPrecondition, "failed to fetch aliases: found SPIRE alias without key: %q", *alias.AliasArn)
+			case alias.AliasArn == nil:
+				return nil, status.Errorf(codes.Internal, "failed to fetch aliases: found SPIRE alias without arn: %q", *alias.AliasArn)
```

Comment:
Constructing this error message will result in a nil pointer dereference since `alias.AliasArn == nil` here. Are there any other useful unique properties of the alias we can include in the error message? `AliasName` (if not also `nil`)?

Reply:
I'll fix this up.